docs/react-components-structure.md

```diff
@@ -0,0 +1,161 @@
+---
+id: react-components-structure
+title: React components structure
+---
+
+In Front Commerce we have seperated our components in two categories: the **UI**
+components available in the `ui` folder, and the **Business** components
+available in the `modules` and `pages` folders.
+
+```
+src/
+├── ...
+└── web/
+    └── theme/
+        ├── ui/
+        |   ├── atoms/
+        |   ├── molecules/
+        |   ├── organisms/
+        |   └── templates/
+        ├── modules/
+        └── pages/
+```
+
+But what does that mean? Where should we put our components?
+
+## TL;DR
+
+* UI components should be highly reusable and only have theming concerns. They
+  are in `src/web/theme/ui/` and organized according the
+  [Atomic Design principles](http://bradfrost.com/blog/post/atomic-web-design/)
+* Business components are components that are used once which contain complex
+  [business logic](https://en.wikipedia.org/wiki/Business_logic). They are
+  splitted in two: the pages entry point and the modules which can be reused in
+  multiple pages.
+
+In order to understand why, let's go through the history of Front-Commerce.
+
+## Step 1: Atomic Design
+
+Back in early 2016, we started with the will to apply the
+[Atomic Design principles](http://bradfrost.com/blog/post/atomic-web-design/).
+Indeed, it is a really good canvas for those who want to put their feet in the
+Design System world. It gave us guidelines that should help to keep our theme
+nice, clean and consistent. We only needed to split our components in 5 folders
+: atoms, molecules, organisms, templates and pages.
+
+http://bradfrost.com/wp-content/uploads/2013/06/atomic-design.png
+
+And overall, it seemed like a perfect fit with React which embraced the ideas of
+components. We believed that by splitting our components in tiny little pieces
+that would assemble together to create whole pages, it would be easy to apply
+those principles. Thus, we ended with this kind of structure:
```

Comment:
Proposal : « We believed that splitting our components in tiny pieces would help us applying those principles. By assembling them together we could create whole pages easily. »